File: src/my_project/easy_problems/from101to150/num_rook_captures.py
from typing import List 

class Solution:
    def numRookCaptures(self, board: List[List[str]]) -> int:
        coordinate_rook = []
        number_pawns = 0
        

        for i in range(8):
            for j in range(8):

                if board[i][j] == "R":
                    coordinate_rook = [i,j]
        flag = False
        for i in range(coordinate_rook[0]-1, -1, -1):
            if board[i][coordinate_rook[1]] != "." and  board[i][coordinate_rook[1]] != "p":
                flag = True 
            if flag == False and board[i][coordinate_rook[1]] == "p":
                number_pawns +=1
                flag = True


        flag = False
        for i in range(coordinate_rook[0] + 1, 8):
            if board[i][coordinate_rook[1]] != "." and  board[i][coordinate_rook[1]] != "p":
                flag = True 
            if flag == False and board[i][coordinate_rook[1]] == "p":
                number_pawns +=1
                flag = True

        flag = False
        for j in range(coordinate_rook[1]-1, -1, -1):
            if board[coordinate_rook[0]][j] != "." and  board[coordinate_rook[0]][j] != "p":
                flag = True 
            if flag == False and board[coordinate_rook[0]][j] == "p":
                number_pawns +=1
                flag = True
            
        flag = False
        for j in range(coordinate_rook[1]+1, 8):
            if board[coordinate_rook[0]][j] != "." and  board[coordinate_rook[0]][j] != "p":
                flag = True 
            if flag == False and board[coordinate_rook[0]][j]== "p":
                number_pawns +=1
                flag = True





        return number_pawns  

solution = Solution()

print(solution.numRookCaptures(board = [[".",".",".",".",".",".",".","."],[".",".",".","p",".",".",".","."],[".",".",".","R",".",".",".","p"],[".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".","."],[".",".",".","p",".",".",".","."],[".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".","."]]))

print(solution.numRookCaptures(board = [[".",".",".",".",".",".",".","."],[".","p","p","p","p","p",".","."],[".","p","p","B","p","p",".","."],[".","p","B","R","B","p",".","."],[".","p","p","B","p","p",".","."],[".","p","p","p","p","p",".","."],[".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".","."]]))

File: src/my_project_typescript/easy_problems_typescript/from1to50/numRookCaptures.ts
function numRookCaptures(board: string[][]): number {



    let coordinateRook: Array<number> = [];
    let numberPawns: number = 0;
    
    for (let i = 0; i< 8; i++){
        for (let j = 0; j<8; j++){
            if (board[i][j]=="R"){
                coordinateRook = [i,j]
            }
        }
    }

    let flag = false;
    for (let i = coordinateRook[0] -1; i > -1; i--){
        if (board[i][coordinateRook[1]] !== "." && board[i][coordinateRook[1]] !== "p"){
            flag = true
        }
        if (flag == false && board[i][coordinateRook[1]] == "p"){
            numberPawns += 1
            flag = true
        }
    }

    flag = false;
    for (let i = coordinateRook[0] +1; i < 8; i++){
        if (board[i][coordinateRook[1]] !== "." && board[i][coordinateRook[1]] !== "p"){
            flag = true
        }
        if (flag == false && board[i][coordinateRook[1]] == "p"){
            numberPawns += 1
            flag = true
        }
    }

    flag = false;
    for (let j = coordinateRook[1] -1; j > -1; j--){
        if (board[coordinateRook[0]][j] !== "." && board[coordinateRook[0]][j] !== "p"){
            flag = true
        }
        if (flag == false && board[coordinateRook[0]][j] == "p"){
            numberPawns += 1
            flag = true
        }
    }

    flag = false;
    for (let j = coordinateRook[1] +1; j < 8; j++){
        if (board[coordinateRook[0]][j] !== "." && board[coordinateRook[0]][j] !== "p"){
            flag = true
        }
        if (flag == false && board[coordinateRook[0]][j] == "p"){
            numberPawns += 1
            flag = true
        }
    }


    return numberPawns
};

console.log(numRookCaptures([[".",".",".",".",".",".",".","."],[".","p","p","p","p","p",".","."],[".","p","p","B","p","p",".","."],[".","p","B","R","B","p",".","."],[".","p","p","B","p","p",".","."],[".","p","p","p","p","p",".","."],[".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".","."]]))
console.log(numRookCaptures([[".",".",".",".",".",".",".","."],[".",".",".","p",".",".",".","."],[".",".",".","R",".",".",".","p"],[".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".","."],[".",".",".","p",".",".",".","."],[".",".",".",".",".",".",".","."],[".",".",".",".",".",".",".","."]]))